tests/tests/tier0/bluechi-reload-unit-service/test_bluechi_reload_unit_service.py

```diff
@@ -0,0 +1,44 @@
+# SPDX-License-Identifier: LGPL-2.1-or-later
+import logging
+import os
+
+from typing import Dict
+from bluechi_test.test import BluechiTest
+from bluechi_test.machine import BluechiControllerMachine, BluechiAgentMachine
+from bluechi_test.config import BluechiControllerConfig, BluechiAgentConfig
+
+LOGGER = logging.getLogger(__name__)
+
+NODE_FOO = "node-foo"
+
+
+def exec(ctrl: BluechiControllerMachine, nodes: Dict[str, BluechiAgentMachine]):
+
+    foo = nodes[NODE_FOO]
+    target_dir = os.path.join("/", "etc", "systemd", "system")
+    foo.copy_systemd_service("service-for-reload.service", "systemd", target_dir)
+    foo.copy_systemd_service("simple.service", "systemd", target_dir)
+    assert foo.wait_for_unit_state_to_be("simple.service", "inactive")
+    assert foo.wait_for_unit_state_to_be("service-for-reload.service", "inactive")
+
+    ctrl.bluechictl.start_unit(NODE_FOO, "service-for-reload.service")
+    assert foo.wait_for_unit_state_to_be("simple.service", "inactive")
+    assert foo.wait_for_unit_state_to_be("service-for-reload.service", "active")
+
+    ctrl.bluechictl.reload_unit(NODE_FOO, "service-for-reload.service")
+    assert foo.wait_for_unit_state_to_be("simple.service", "active")
+    assert foo.wait_for_unit_state_to_be("service-for-reload.service", "active")
+
+
+def test_agent_config_c_option(
```

Comment:
Please rename the fuction to match module name: `test_bluechi_reload_unit_service`